### PR TITLE
nocomma: fix false positives

### LINKED ISSUE #2044: "nocomma" shouldn't apply to array/object literals and parameter lists
Introduced in #1876, this looks broken. It's emitting warnings for commas in e.g. object literals. I found this by enabling the new `enforceall` option, which is practically unusable for any serious project as long as it includes `nocomma` in its current state.

```json
{
  "nocomma": true
}
```
```js
function foo(d, e) {
	return {
		a: 1,
		b: 2,
		c: ['x', 'y'],
		d: e,
		e: e
	};
}
foo();
```

```
      1 |function foo(d, e) {
                       ^ Unexpected use of a comma operator.
      3 |        a: 1,
                     ^ Unexpected use of a comma operator.
      4 |        b: 2,
                     ^ Unexpected use of a comma operator.
      5 |        c: ['x', 'y'],
                        ^ Unexpected use of a comma operator.
      5 |        c: ['x', 'y'],
                              ^ Unexpected use of a comma operator.
      6 |        d: e,
                     ^ Unexpected use of a comma operator.
```

### PR DESCRIPTION
**Pull request by denis-sokolov**
Fix #2044.

### CODE REVIEW

**Comment by impinball** [comment]
@denis-sokolov You mind fixing this (hopefully) last nit by @rwaldron?

**Comment by denis-sokolov** [comment]
@impinball, I have already received the notification about that comment from GitHub, I really don't need a second one. Thank you.

**Comment by impinball** [comment]
Sorry.

**Comment by denis-sokolov** [comment]
@rwaldron, fixed the quote style.

File: tests/unit/options.js [hunk 1]
```diff
@@ -1676,17 +1676,35 @@ exports.nonbsp = function (test) {
 /** Option `nocomma` disallows the use of comma operator. */
 exports.nocomma = function (test) {
   // By default allow comma operator
-  TestRun(test)
+  TestRun(test, "nocomma off by default")
     .test("return 2, 5;", {});
```
Comment (caitp) [inline tests/unit/options.js]: something I don't get: we're not complaining about return statements being used outside of functions, which is an early error everywhere

Comment (rwaldron) [inline tests/unit/options.js]: I've been thinking about the same question, but had dismissed it as "it's an early error anyway". Do you think we should be more aggressive there?

Comment (caitp) [inline tests/unit/options.js]: maybe it's fine the way it is

**Comment by rwaldron** [comment]
Please update the commit message to follow https://github.com/jshint/jshint/blob/master/CONTRIBUTING.md#commit-message-guidelines Thanks!!

**Comment by rwaldron** [comment]
@denis-sokolov I will land as soon as you update the commit message :)

**Comment by denis-sokolov** [comment]
@rwaldron, does this work?

**Comment by jugglinmike** [comment]
@denis-sokolov I'd like to merge this, but recent changes to `master` have introduced conflicts in your patch. They are trivial--just code style changes--so I would be happy to resolve them on your behalf when I squash the commits. Let me know if that's okay, or if you would prefer to do it yourself

**Comment by denis-sokolov** [comment]
@jugglinmike, done.

**Comment by jugglinmike** [comment]
Squashed and merged to `master` at commit 33612f8cedce779d0d22571b72da4216d18bee9c . Thanks for your patience on this one, Denis!

**Pull request closed without merging**